packages/flutter_tools/lib/src/version.dart

```diff
@@ -16,19 +16,25 @@ import 'globals.dart' as globals;
 
 const String _unknownFrameworkVersion = '0.0.0-unknown';
 
+/// This maps old branch names to the names of branches that replaced them.
+///
+/// For example, in 2021 we deprecated the "dev" channel and transitioned "dev"
+/// users to the "beta" channel.
+const Map<String, String> kObsoleteBranches = <String, String>{
+  'dev': 'beta',
```

Comment:
might as well add alpha here too (also going to beta), for completeness...

Reply:
i guess maybe not, we'd also have to add hackathon and codelab and at some point there's a question of how much we care to have all these on master, since they only matter on the actual branch going away. In fact, if anything, this can literally just be a single pair, since there will never be a case where this needs to be more than one thing.